### PR TITLE
adding datastore models for ner.

### PR DESCRIPTION
**Pull request by danyaljj**
Using datastore to load NER models.

### CODE REVIEW

File: ner/src/main/java/edu/illinois/cs/cogcomp/ner/config/NerBaseConfigurator.java [hunk 1]
```diff
@@ -35,6 +35,10 @@
     public final static String PATH_TO_MODEL = "pathToModelFile";
     public final static String VIEW_NAME = "viewName";
 
+    public final static String TRAINED_ON = "all-data";
```
Comment (danyaljj) [inline ner/src/main/java/edu/illinois/cs/cogcomp/ner/config/NerBaseConfigurator.java]: To choose whether we use the models that are train on: 
- all data 
or 
- training data only

File: ner/src/main/java/edu/illinois/cs/cogcomp/ner/config/NerBaseConfigurator.java [hunk 2]
```diff
@@ -130,6 +134,7 @@ public ResourceManager getDefaultConfig() {
         Properties props = new Properties();
 
         props.setProperty(VIEW_NAME, ViewNames.NER_CONLL);
+        props.setProperty(TRAINED_ON, TRAINED_ON_ALL_DATA);
```
Comment (danyaljj) [inline ner/src/main/java/edu/illinois/cs/cogcomp/ner/config/NerBaseConfigurator.java]: the default is "models trained on all data"

File: ner/src/main/java/edu/illinois/cs/cogcomp/ner/NERAnnotator.java [hunk 3]
```diff
@@ -100,20 +106,37 @@ public void initialize(ResourceManager nerRm) {
...
+        else {
+            dataset = "ontonotes";
+        }
+
+        String data_split = nerRm.getString(NerBaseConfigurator.TRAINED_ON);
+
+        try {
+            Datastore ds = new Datastore(new ResourceConfigurator().getDefaultConfig());
+            String artifact_id = "ner-model-" + dataset + "-" + data_split;
+            File modelDir = ds.getDirectory("edu.illinois.cs.cogcomp.ner", artifact_id, 4.0, false);
```
Comment (danyaljj) [inline ner/src/main/java/edu/illinois/cs/cogcomp/ner/NERAnnotator.java]: load the model from datastore here.

File: ner/src/main/java/edu/illinois/cs/cogcomp/ner/NERAnnotator.java [hunk 4]
```diff
@@ -100,20 +106,37 @@ public void initialize(ResourceManager nerRm) {
...
+            dataset = "enron-conll";
+        }
+        else {
+            dataset = "ontonotes";
+        }
+
+        String data_split = nerRm.getString(NerBaseConfigurator.TRAINED_ON);
+
+        try {
+            Datastore ds = new Datastore(new ResourceConfigurator().getDefaultConfig());
```
Comment (cogcomp-dev) [inline ner/src/main/java/edu/illinois/cs/cogcomp/ner/NERAnnotator.java]: Shouldn't this use the ResourceManager argument provided to the constructor to overwrite any properties the user wants to configure?

File: ner/src/main/java/edu/illinois/cs/cogcomp/ner/config/NerBaseConfigurator.java (hunk 1 above)
Comment (cogcomp-dev) [reply ner/src/main/java/edu/illinois/cs/cogcomp/ner/config/NerBaseConfigurator.java]: Nice.

File: ner/src/main/java/edu/illinois/cs/cogcomp/ner/NERAnnotator.java (hunk 4 above)
Comment (danyaljj) [reply ner/src/main/java/edu/illinois/cs/cogcomp/ner/NERAnnotator.java]: Good point. Updated it.

**Pull request merged**